org.osgi.test.cases.jdbc/src/org/osgi/test/cases/jdbc/junit/JDBCTestCase.java

```diff
@@ -57,127 +74,155 @@ private static class DfltProps extends Properties {
-		assertThat(sa.getService()).isNotNull();
-
-		ServiceReferenceAssert.assertThat(sa.getServiceReference())
-				.hasServicePropertiesThat()
-				.containsKey(OSGI_JDBC_CAPABILITY)
-				.extractingByKey(OSGI_JDBC_CAPABILITY)
-				.isNotNull();
-
-	}
-
+	private static final List<JdbcCapability<?>> ALL_CAPABILITIES = List.of(
+			new JdbcCapability<Driver>(OSGI_JDBC_CAPABILITY_DRIVER, DataSourceFactory::createDriver, 
+					"org.osgi.service.jdbc.DataSourceFactory.createDriver(Properties)", Properties::new), 
+			new JdbcCapability<DataSource>(OSGI_JDBC_CAPABILITY_DATASOURCE, DataSourceFactory::createDataSource, 
+					"org.osgi.service.jdbc.DataSourceFactory.createDataSource(Properties)", DfltProps::new), 
+			new JdbcCapability<ConnectionPoolDataSource>(OSGI_JDBC_CAPABILITY_CONNECTIONPOOLDATASOURCE, DataSourceFactory::createConnectionPoolDataSource, 
+					"org.osgi.service.jdbc.DataSourceFactory.createConnectionPoolDataSource(Properties)", DfltProps::new), 
+			new JdbcCapability<XADataSource>(OSGI_JDBC_CAPABILITY_XADATASOURCE, DataSourceFactory::createXADataSource, 
+					"org.osgi.service.jdbc.DataSourceFactory.createXADataSource(Properties)", DfltProps::new)
+		);
+
+	@InjectBundleContext
+	BundleContext bundleContext;
+	
+	/**
+	 * This test that 
+	 * <ul>
+	 * <li>all required properties are declared and of the required type</li>
+	 * <li>all optional properties are of the required type</li>
+	 * </ul>
+	 * @param sr the service reference of the {@link DataSourceFactory}
+	 */
 	@ParameterizedTest
 	@ServiceSource(serviceType = DataSourceFactory.class)
-	public void testRegisteredProperties(
-			ServiceReference<DataSourceFactory> sr) {
+	public void testRegisteredProperties(ServiceReference<DataSourceFactory> sr) {
 
 		ServiceReferenceAssert<DataSourceFactory> sRefAssert = ServiceReferenceAssert
 				.assertThat(sr);
 
+		//first check all mandatory properties
 		sRefAssert.hasServicePropertiesThat()
-				.as("The DataSourceFactory is missing the required osgi.jdbc.driver.class property")
+				.as("According to '125.2 Database Driver' the DataSourceFactory must specify the '%s' with its registration.",
+						OSGI_JDBC_DRIVER_CLASS)
 				.containsKey(OSGI_JDBC_DRIVER_CLASS)
 				.extractingByKey(OSGI_JDBC_DRIVER_CLASS)
 				.isNotNull()
-				.as("The DataSourceFactory driver class is not a String")
+				.as("According to '125.2 Database Driver' the property '%s' must be a String", OSGI_JDBC_DRIVER_CLASS)
 				.isInstanceOf(String.class);
+		
+		sRefAssert.hasServicePropertiesThat()
+				.as(String.format("According to '125.2 Database Driver' the DataSourceFactory must specify the '%s' with its registration.",
+						OSGI_JDBC_CAPABILITY))
+				.containsKey(OSGI_JDBC_CAPABILITY)
+				.extractingByKey(OSGI_JDBC_CAPABILITY)
+				.isNotNull()
+				.as("According to '125.2 Database Driver' the property '%s' must be a String+", OSGI_JDBC_CAPABILITY)
+				.isInstanceOfAny(String.class, String[].class);
```

Comment:
String+ includes a collection of String.

    For example String+, indicates that a String, a String[], and a Collection<String> are all valid forms.